sympy/stats/joint_rv.py

```diff
@@ -0,0 +1,170 @@
+"""
+Joint Random Variables Module
+
+See Also
+========
+sympy.stats.rv
+sympy.stats.frv
+sympy.stats.crv
+sympy.stats.drv
+"""
+
+from __future__ import print_function, division
+
+__all__ = ['Joint']
+
+from sympy import Basic, Lambda, sympify
+from sympy.core.symbol import Symbol
+from sympy.core.compatibility import string_types
+from sympy.concrete.summations import Sum
+from sympy.integrals.integrals import Integral
+from sympy.stats.rv import (ProductPSpace, RandomSymbol, pspace,
+        NamedArgsMixin, ProductDomain)
+
+class JointPSpace(ProductPSpace):
+    """
+
+    """
+    def __new__(cls, name, syms, dist, **args):
```

Comment:
`name` is same as the symbol used to initialize the object, same as what is done for other random variables. Eg: In `Y = Poisson('Y', 1)`, 'Y' is analogous to `name`.

Reply:
`RandomSymbol` supports name. I don't think the product space class should have a name.

Reply:
`SinglePSpace` takes a symbol as one of the arguments, which is the same that is used for the initialisation of the random symbol. Should the same not be done for the joint space?

Reply:
It seems to me that a joint space does not have a single distinguished random variable whose name would describe the whole space unlike `SinglePSpace`.

Reply:
I believe the symbol taken as argument to `SinglePSpace` is needed for independence (in case of more pspaces with the same distribution). You are already passing `syms`, right? Aren't `syms` already identifying the joint probability space?

Reply:
Oh okay. I'll make the correction.